### PR TITLE
Demo-related improvements.

### PR DESCRIPTION
**Pull request by gtopper**
* Allow sparse data in writers, and handle it appropriately in each writer.
* Rename EmitBase -> EmitPolicy
* Raise type error when EmitPolicy is of the wrong type (such as when `EmitEveryEvent` is passed in instead of `EmitEveryEvent()`).
* Report the offending update expression on error where relevant.

### CODE REVIEW

File: storey/drivers.py [hunk 1]
```diff
@@ -158,7 +160,9 @@ async def _save_key(self, container, table_path, key, aggr_item, partitioned_by_
 
         if should_raise_error:
             raise V3ioError(
-                f'Failed to save aggregation for {table_path}/{key}. Response status code was {response.status_code}: {response.body}')
+                f'Failed to save aggregation for {table_path}/{key}. Response status code was {response.status_code}: {response.body}\n'
+                f'Update expression was: {update_expression}'
```
Comment (dinal) [inline storey/drivers.py:164]: why did you remove the update expr? it's easier to debug with it

Comment (gtopper) [reply storey/drivers.py:164]: That's why I added it, not remove it 🤔

**Review by katyakats — APPROVED** [review]

File: storey/drivers.py (hunk 1 above)
Comment (dinal) [reply storey/drivers.py:164]: lol :D
